prow/pod-utils/gcs/upload.go

```diff
@@ -52,18 +53,23 @@ func LocalExport(exportDir string, uploadTargets map[string]UploadFunc) error {
 			filePath: path.Join(exportDir, dest),
 		}
 	}
-	return upload(dtw, uploadTargets)
+	return upload(dtw, uploadTargets, int64(len(uploadTargets))) // there is no use in limiting concurrency of local writes
 }
 
-func upload(dtw destToWriter, uploadTargets map[string]UploadFunc) error {
+func upload(dtw destToWriter, uploadTargets map[string]UploadFunc, maxConcurrency int64) error {
 	errCh := make(chan error, len(uploadTargets))
 	group := &sync.WaitGroup{}
 	group.Add(len(uploadTargets))
+	sem := semaphore.NewWeighted(maxConcurrency)
 	for dest, upload := range uploadTargets {
 		log := logrus.WithField("dest", dest)
+		if err := sem.Acquire(context.Background(), 1); err != nil {
```

Comment:
Note: it's actually the `dtw(dest)` call we make in creating the closure that allocates, so we need to guard _that_